ext/mbstring/libmbfl/filters/mbfilter_cp5022x.c

```diff
@@ -589,16 +578,6 @@ mbfl_filt_conv_wchar_cp50220_ctor(mbfl_convert_filter *filt)
 	vtbl_tl_jisx0201_jisx0208.filter_ctor(filt);
 }
 
-static void
-mbfl_filt_conv_wchar_cp50220_copy(mbfl_convert_filter *src, mbfl_convert_filter *dest)
-{
-	mbfl_filt_conv_wchar_cp50220_ctx *ctx;
-
-	*dest = *src;
-	ctx = emalloc(sizeof(mbfl_filt_conv_wchar_cp50220_ctx));
-	dest->data = ctx;
-}
```

Comment:
In the mime implementation, you now explicitly back up and restore status & cache. Could there be a problem with filters like this one, that have additional state beyond that?

Reply:
Will look into this.

Reply:
@nikic, the filter you are looking at there doesn't really store additional state. When you create a filter for CP50220, the filter constructor just inserts another filter in front of itself in the chain. Then it has some flags which don't change as it runs through an input string, and which therefore don't need to be saved and restored when doing MIME encoding.

I don't want to jump to the conclusion that my code is automatically correct, but in this case, there doesn't seem to be any other state to save and restore.

The one filter which definitely *does* have additional state is the one for HTML entity encoding and decoding. Which is actually weird, because it means we have code for HTML entity encoding/decoding *both* in mbfilter.c and in mbfilter_htmlent.c. That's not just me who duplicated it; when I started working on mbstring, two implementations were already there.

Anyways, if the input encoding for text which needs to be MIME-encoded is "HTML entity-encoded", then definitely we would have a problem. Maybe I should work on that. Sounds pretty crazy, though.

Reply:
Another weird thing about mbstring is that some of the text "encodings" which it supports are things like "QPrint-encoding" and "uuencode", which are not really text encodings but just alternative ways of encoding raw bytes.